### DISCUSSION TITLE
Trigger validation after initial data is loaded

### DISCUSSION THREAD

**Discussion by Marcin-Kapusta** [react-hook-form/react-hook-form#3700] [General]
I would like to validate form as soon as initial data is loaded.

To set loaded data I'm using reset and to trigger validation I'm trying to use trigger()
In the result no errors are showing. If I wait for example 100ms and then call trigger then validation is working. What I'm doing wrong. 

Also when defaultValues are set during useForm then trigger works as expected but in my case I want to simulate async loaded data so I'm using reset in the example below.

Here is the Sandbox: https://codesandbox.io/s/reack-hook-form-initial-validation-zrwxv

Cheers :)

**Comment by bluebill1049** [comment]
`trigger` needs to happen after `reset` is completed, 

- `reset`
- re-render the form populated inputs
- then `trigger`

**Reply by Marcin-Kapusta** [reply]
How to wait for reset to be completed. There is no Promise Result from calling this method?

What do You mean `re-render the form populated inputs`? Do You have some example in sandbox?

**Reply by bluebill1049** [reply]
here is a an example: https://codesandbox.io/s/reack-hook-form-initial-validation-forked-9zmqt

**Reply by ssaric** [reply]
This is a Really useful case. Could we add this to the example somewhere?

**Reply by aditya-lenze** [reply]
In my case, I was get it to work by following the above CSB, but only by surrounding the `trigger` with a `setTimeout` of a small value. 
Just using `trigger` does not show any errors with the initially loaded data.
RHF version: 7.52.2.

**Reply by breadcrumbsonthefloor** [reply]
> here is a an example: https://codesandbox.io/s/reack-hook-form-initial-validation-forked-9zmqt

this example does not work anymore, after setIsReset to true, the following useEffect containing the if (isReseted) will have the previous methods.formState.defaultValues instead of the newer ones, a way to fix it now is to include form.formState.defaultValues as dependency, but we don't know when the reset happen, so this will produce undesired effect of the field validation triggering twice

On version 7.51.2 and 7.55.0
